### DISCUSSION TITLE
Android app crashes when tapping a button

### DISCUSSION THREAD

**Discussion by jeromelaban** [unoplatform/uno#9516] [Q&A]
[_From  luciusinfabula on Discord:_](https://discord.com/channels/372137812037730304/550416151172087808/1009038610944770098)

Hi all. I just started learning Uno Platform and I recreated a project starting from this one: https://github.com/pmbanugo/MyWasmApp.SPA. I Created new solution: https://github.com/luciusinfabula/MyApp3 including the other platforms because the original project has only Wasm template. The project works perfectly for Uwp, Wpf host and Wasm but not in Droid. In Android the app starts but doesn't show the window correctly and crashes when I tap many times Add button. I'm using Visual Studio 2019 and a real device Asus Zenphone 3 with Android 8.0.0. Someone could help me to understand what I'm doing wrong?

**Comment by jeromelaban** [comment]
This looks unusual. Could you share the exception stack trace of the crash?

**Comment by luciusinfabula** [comment]
Hi Jerome,
I'm not able to catch the exception at the moment of the crash, the app both at debug or running standalone simply shows the message: MyApp3 has stopped, then closes.
At debug, I tried clicking F11 when tap on Add button but the steps are hundreds and I cannot catch the exact moment in which the problem occurs.
When the app starts correctly in WPF, UWP and WASM this is the screenshot:
![Screenshot_Uwp](https://user-images.githubusercontent.com/54881305/185776369-4bbf5427-7faf-4394-adc2-6b0fd85b0bc5.jpg)
While opening in Android shows this one:
![Screenshot_Android](https://user-images.githubusercontent.com/54881305/185776376-a34a56c3-4acd-4675-a7b3-fcfc69c17f2c.jpg)

Luciano

**Reply by jeromelaban** [reply]
This is unfortunately not enough to help you there. You may be able to find the crash details in the device logs, or in the visual studio output window.

**Comment by luciusinfabula** [comment]
Hi Jerome,
sorry for the delay of answer.
Following your hint I saw output messages in Visual Studio and noted a serie of messages like following:

> [OpenGLRenderer] Shape too large to be rendered into a texture (-2147483644x138, max=16384x16384)

Then tried reducing grids and stackpanels width and now the screen appears like that:

![Screenshot](https://user-images.githubusercontent.com/54881305/187831097-865add8c-5ff8-4edf-a01b-1e0d42a16c4f.jpg)

Now it is possible to tapp Add button whenever you need without causing exceptions or app crash; it was simple a problem about window sizing.

I have to figure out how to size correctly the window but I consider the issue to be closed.

Thanks again for your help.

Luciano

**Reply by jeromelaban** [reply]
Thanks for the update. Which version of android are you running on?

**Reply by jeromelaban** [reply]
_(make sure to keep answering to the same conversation, even if GitHub's UI is confusing :) )_ 

Thanks for the version. I recall that earlier version of Android had such size limitations, though in general panels and other items should be be generating large views like that. Would you have an example of modifications you made to make this work?

**Reply by luciusinfabula** [reply]
Hi Jerome, sorry for the delay but I had no time previously to get a real solution to the issue.

The problem was with this xaml piece of code inside the tag Grid.ColumnDefinitions:

            <ColumnDefinition Width="Auto"/>
            <ColumnDefinition Width="Auto"/>


that worked correctly in WPF and UWP but not in Droid because of app crashing.

One first solution was this;

            <ColumnDefinition Width="150"/>
            <ColumnDefinition Width="150"/>

but it's not satisfying because doesn't resize correctly, very ugly to see.

The best one to me is the following:

            <ColumnDefinition Width="*"/>
            <ColumnDefinition Width="*"/>

that both works ok and is beautiful for UI representation, resizing, rotating, etc.

Bye.
Luc
